presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MetadataDao.java

```diff
@@ -276,9 +278,28 @@ long insertDistribution(
     @SqlQuery("SELECT table_id\n" +
             "FROM tables\n" +
             "WHERE organization_enabled\n" +
+            "  AND maintenance_blocked IS NULL\n" +
             "  AND table_id IN\n" +
             "       (SELECT table_id\n" +
             "        FROM columns\n" +
             "        WHERE sort_ordinal_position IS NOT NULL)")
     Set<Long> getOrganizationEligibleTables();
+
+    @SqlUpdate("UPDATE tables SET maintenance_blocked = CURRENT_TIMESTAMP\n" +
+            "WHERE table_id = :tableId\n" +
+            "  AND maintenance_blocked IS NULL")
+    void blockMaintenance(@Bind("tableId") long tableId);
+
+    @SqlUpdate("UPDATE tables SET maintenance_blocked = NULL\n" +
+            "WHERE table_id = :tableId")
+    void unblockMaintenance(@Bind("tableId") long tableId);
+
+    @SqlQuery("SELECT maintenance_blocked IS NOT NULL\n" +
+            "FROM tables\n" +
+            "WHERE table_id = :tableId\n" +
+            "FOR UPDATE")
+    boolean isMaintenanceBlockedLocked(@Bind("tableId") long tableId);
+
+    @SqlUpdate("UPDATE tables SET maintenance_blocked = NULL")
```

Comment:
It uses the index if you add a `WHERE maintenance_blocked IS NOT NULL`. Otherwise it doesn't (lame).